### ISSUE TITLE
autoPropertyToField broken for interfaces

### ISSUE THREAD

**Issue by Danielku15** [bridgedotnet/Bridge#770]
There seems to be an issue with the `autoPropertyToField` option when using interfaces. It fails with some overload/name conflict error. I perpared a small example to reproduce the issue. 

### Example 
**bridge.json**

    {
      "output": "Bridge/output",
      "autoPropertyToField": true
    }

**Test.cs**

    namespace BridgeCompileFail
    {
        public interface IBase
        {
            float Prop { get; set; }
        }

        public class Impl : IBase
        {
            public float Prop { get; set; }
        }
    }

### Compile output
	Test.cs(9,6,12,7): error : Error: Cannot translate interface (BridgeCompileFail.IBase) member '[Property BridgeCompileFail.Impl.Prop]' in 'BridgeCompileFail.Impl' due name conflicts. Please rename methods or refactor your code    at Bridge.Contract.OverloadsCollection.GetOverloadName(IMember definition)
	Test.cs(9,6,12,7): error :    at Bridge.Contract.OverloadsCollection.GetOverloadName()
	Test.cs(9,6,12,7): error :    at Bridge.Contract.TypeConfigItem.GetName(IEmitter emitter)
	Test.cs(9,6,12,7): error :    at Bridge.Translator.ClassBlock.<EmitClassHeader>b__2(TypeConfigItem m)
	Test.cs(9,6,12,7): error :    at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
	Test.cs(9,6,12,7): error :    at Bridge.Translator.ClassBlock.EmitClassHeader()
	Test.cs(9,6,12,7): error :    at Bridge.Translator.ClassBlock.DoEmit()
	Test.cs(9,6,12,7): error :    at Bridge.Translator.AbstractEmitterBlock.Emit()
	Test.cs(9,6,12,7): error :    at Bridge.Translator.EmitBlock.DoEmit()
	Test.cs(9,6,12,7): error :    at Bridge.Translator.AbstractEmitterBlock.Emit()
	Test.cs(9,6,12,7): error :    at Bridge.Translator.Emitter.Emit()
	Test.cs(9,6,12,7): error :    at Bridge.Translator.Translator.Translate()
	Test.cs(9,6,12,7): error :    at Bridge.Build.GenerateScript.Execute()

[Edit 1]
I debugged into the OverloadsCollection and the error has to do with the change of the property name casing. While for `Impl` the property is changed to from `Prop` to `prop` the casing is not changed for `IBase.Prop`. This is because the `autoPropertyToField` option only applies for auto properties and the interface property is not detected as auto property. 

Basically the error is: The compiler wants to emit `Impl.Prop` as simple field due to the `autoPropertyToField`, but the `autoPropertyToField` option must be ignored if the auto property implements a interface member.

**Comment by Danielku15** [comment]
I prepared a pull request that solves the issue on my local environment (#771). 

I disabled the `autoPropertyToField` feature completely for properties that implement interface members since it breaks the compilation.

**Comment by DaniilVeriga** [comment]
@Danielku15, thank you for the report and pull request. 
@vladsch, could you, please, review?

**Comment by vladsch** [comment]
Unfortunately, exception in this case is desired behaviour. Interface property can be implemented like auto property or like property with getter/setter in a class 
Access to auto and custom property is different (if auto property is converted to a field). I mean that field or methods will be used to read/write the property in Javascript 
But if you assign class instance to interface variable (like, IFoo foo = new Class();) then translator does not know how to emit access to the property (if property is used via interface variable), information about instance type which is referenced in interface variable is not available during translation
So, if you sure that interface property will be always auto in classes then apply FieldProperty attribute to interface property

**Comment by Danielku15** [comment]
Taking polymorphism into account you can never be sure if an interface member is implemented as auto property or not. 

But I agree that if the developer is aware of the side effect regarding issues and `autoPropertyToField` I would say a second option `interfacePropertyToField` forcing interfaces properties into field would be the better solution. 

I don't want to spoil my whole code with bridge.net attributes since I'm developing a cross platform library that is not only targeting JavaScript from the codebase. Fulling up my abstraction layer (interfaces) with preprocessors and attributes is somehow not clean to me. I'd prefer a compiler option or assembly level attributes (like SharpKit) to control my platform specific code. 

If we can agree on a solution I will try to implement . Here some proposals:

* Ignore autoPropertyToField for interface members (like the current pull request)
* Force all interface properties to fields if autoPropertyToField is set + a compiler warning/error if a interface member is not implemented as auto property but the option is enabled for it.
* A new option interfacePropertyToField which forces all interface properties to be compiled to fields + a compiler error if a interface member is not implemented as auto property.

**Comment by vladsch** [comment]
Thanks for your proposals, i will discuss it with the team

**Comment by LeonidVeriga** [comment]
@Danielku15,

Thank you for the Pull Request. It is very good and has been merged.

We have also added a Compiler test added by 86887e7
